packages/components/view-switcher/src/view-switcher.tsx

```diff
@@ -16,28 +17,44 @@ type TReactChild = {
 
 export type TViewSwitcherProps = {
   /**
-   * Indicates that the view switcher can be reduced to save space
+   * Indicates that the view switcher can be reduced to save space.
    */
   isCondensed?: boolean;
   /**
-   * Pass one or more `ViewSwitcher.Button` components
+   * Pass one or more `ViewSwitcher.Button` components.
    */
   children: ReactNode;
   /**
-   * Will be triggered whenever a `ViewSwitcher.Button` is selected. Called with the ViewSwitcherButton value
+   * Will be triggered whenever a `ViewSwitcher.Button` is selected. Called with the ViewSwitcherButton value.
+   * This function is only required when the component is controlled.
    */
   onChange?: (value: string) => void;
   /**
-   * Indicates the default selected button
+   * Passing this prop makes the component an uncontrolled component.
+   * Indicates the default selected button it is only used to as an initial state once, when the component mounts.
    */
-  defaultSelected: string;
+  defaultSelected?: string;
+  /**
+   * Passing this prop makes the component an controlled component.
+   * Controlled components also require to pass a `onChange` callback function.
+   */
+  selectedValue?: string;
 };
 
 const ViewSwitcher = (props: TViewSwitcherProps) => {
-  const [selectedButton, setSelectedButton] = useState<string>(
+  const isControlledComponent = !isNil(props.selectedValue);
+  const hasOnChange = !isNil(props.onChange);
+  const [selectedButton, setSelectedButton] = useState<string | undefined>(
```

Comment:
Could be a needless useState but if we want to avoid it we need two child components and increase complexity due to the cloning possibly needing to be shared. Let me know what you think.

Reply:
Yeah, I think it's fine yes. The state is only used anyway in uncontrolled mode.

Maybe let's add a comment to make it explicit.